lib/blocks.php

```diff
@@ -147,6 +150,46 @@ function gutenberg_reregister_core_block_types() {
 
 add_action( 'init', 'gutenberg_reregister_core_block_types' );
 
+/**
+ * Registers block styles for a core block.
+ *
+ * @param string $block_name The block-name.
+ *
+ * @return void
+ */
+function gutenberg_register_core_block_styles( $block_name ) {
+	if ( ! gutenberg_should_load_separate_block_styles() ) {
+		return;
+	}
+
+	$block_name = str_replace( 'core/', '', $block_name );
+
+	$style_path        = is_rtl()
+		? "build/block-library/blocks/$block_name/style-rtl.css"
+		: "build/block-library/blocks/$block_name/style.css";
+	$editor_style_path = is_rtl()
+		? "build/block-library/blocks/$block_name/style-editor-rtl.css"
+		: "build/block-library/blocks/$block_name/style-editor.css";
+
+	if ( file_exists( gutenberg_dir_path() . $style_path ) ) {
+		wp_register_style(
+			'wp-block-' . $block_name,
```

Comment:
There is `generate_block_asset_handle` function:
https://developer.wordpress.org/reference/functions/generate_block_asset_handle/

that we can use here and in the second place to generate the handle that will remain the same when we refactor to use file paths. I'm aware it's a lot of work updating `block.json` files so we can tackle it as a follow-up PR 😄

Reply:
I actually like the pattern you used for style handles more than what I encoded in `generate_block_asset_handle`. I'm inclined to change it in WordPress Core instead 😄

Reply:
How about we handle it this way: https://github.com/WordPress/wordpress-develop/pull/820 to ensure that all style handles provided for core blocks would match with what `generate_block_asset_handle` will generate in the future when we migrate to local files?

Reply:
Makes perfect sense to me! :+1: